WordPress/Classes/ViewRelated/Gutenberg/Layout Picker/GutenbergLayoutPickerViewController.swift

```diff
@@ -255,20 +315,59 @@ extension GutenbergLayoutPickerViewController: UITableViewDelegate {
             self.headerView.layoutIfNeeded()
         }, completion: nil)
     }
+
+    private func containsSelectedLayout(_ layout: GutenbergSelectedLayout, atIndexPath indexPath: IndexPath) -> Bool {
+        let rowSection = sections[indexPath.row]
+        return (layout.sectionSlug == rowSection.section.slug)
+    }
 }
 
 extension GutenbergLayoutPickerViewController: UITableViewDataSource {
 
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return 20 // This is a random number chossen for testing
-    }
-
-    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
-        return 318 // This is an estimated number for the height of the eventual cells. Being used for testing right now.
+        return sections.count
     }
 
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = UITableViewCell(style: .default, reuseIdentifier: nil)
+        let cell = tableView.dequeueReusableCell(withIdentifier: LayoutPickerSectionTableViewCell.cellReuseIdentifier, for: indexPath) as! LayoutPickerSectionTableViewCell
```

Comment:
Could we guard this cast?
I understand this should never fail but it might if we modify this class in the future.

- If we do want to crash let's do it explicitly with a `fatalError` with a descriptive message.
- Alternatively we could try to recover instantiating the cell manually and calling an `assertionFailure` to let us know something is wrong.

Same on the CollectionViewController.

Reply:
Yeah, that's a good point. I think here a guard would just return an empty Table View cell and a crash would be the result of failing to register the correct xib. So I think I like your first option of using the `fatalError` better that will help future developers if something goes wrong.